### PR TITLE
Add optional config flag to choose evaluator for Object Detection

### PR DESCRIPTION
**Pull request by consoull**
### Linked issue(s):
Towards KOL-3869

### What change does this PR introduce and why?
Adds an optional boolean to the evaluator configuration used in the prebuilt Object Detection workflow. The flag can be used to explicitly determine if the SingleClass or Multiclass evaluator is run. In the absence of this flag, the evaluator used is determined dynamically based on the number of distinct classification labels found in the ground truths and inferences.

### Please check if the PR fulfills these requirements

- [x] Include reference to internal ticket and/or GitHub issue "Fixes #NNNN" (if applicable)
- [x] Relevant tests for the changes have been added
- [x] Relevant docs have been added / updated

### CODE REVIEW

File: tests/unit/_experimental/object_detection/test_configurations.py [hunk 1]
```diff
@@ -80,7 +86,7 @@ def test__object_detection__multiple_configurations__multiclass() -> None:
 def test__object_detection__multiple_configurations__single_class() -> None:
     ground_truth = GroundTruth(bboxes=[LabeledBoundingBox(top_left=(0, 0), bottom_right=(1, 1), label="a")])
     inference = Inference(
-        bboxes=[ScoredLabeledBoundingBox(top_left=(0, 0), bottom_right=(1, 1), label="b", score=random.random())],
+        bboxes=[ScoredLabeledBoundingBox(top_left=(0, 0), bottom_right=(1, 1), label="a", score=random.random())],
```
Comment (ly-kolena) [inline tests/unit/_experimental/object_detection/test_configurations.py]: is this change needed?

Comment (consoull) [reply tests/unit/_experimental/object_detection/test_configurations.py]: Not needed but the test was inaccurate. This isn't single class since there is `a` and `b`. I can remove it but figure it's overkill to have it's own pr

**Review by gordonhart — APPROVED** [review]

File: tests/unit/_experimental/object_detection/test_configurations.py (hunk 1 above)
Comment (marko-polo-cheno) [reply tests/unit/_experimental/object_detection/test_configurations.py]: The reason for the inaccuracy is to make sure that the single class evaluator doesn't care about class names

**Review by marko-polo-cheno — APPROVED** [review]
LGTM! Left one note on one test for clarity

File: tests/unit/_experimental/object_detection/test_configurations.py (hunk 1 above)
Comment (consoull) [reply tests/unit/_experimental/object_detection/test_configurations.py]: ah got it. I'll revert this portion
